lib/tasks/hdm.rake

```diff
@@ -18,27 +19,44 @@ namespace :hdm do
   end
 
   desc 'Manual Import of FHIR resources'
-  task :load, %i[file] => :environment do |_t, args|
+  task :load, %i[file create_user account_email account_password] => :environment do |_t, args|
     bundle_json = File.open(args.file, 'r:UTF-8').read
-    users = User.all.to_a
-
-    providers = Provider.all.to_a
+    json_format = JSON.parse(bundle_json)
     user_indx = 0
-    profile_indx = 0
-    provider_indx = 0
-
-    if users.empty?
-      puts 'There are no users in the system, please add a user before continuing'
-      return
-    elsif users.length > 1
-      puts 'Select which user you want to load the data for '
-      users.each_with_index { |u, i| puts "#{i}. #{u.email}" }
-      user_indx = STDIN.gets.chomp.to_i
+    if args.create_user == 'y'
+      patient_entry = json_format['entry'].find \
+       { |eachentry| eachentry.key?('resource') && eachentry['resource'].key?('resourceType') && eachentry['resource']['resourceType'] == 'Patient' }
+      user_first_name = patient_entry['resource']['name'][0]['given'][0]
+      user_last_name = patient_entry['resource']['name'][0]['family']
+      user_email = args.account_email || user_first_name + '_' + user_last_name[0] + '@test.com'
+      user_password = args.account_password || 'Password123'
+      created = User.create(first_name: user_first_name, last_name: user_last_name, email: user_email, password: user_password)
+      if !created.validate
+        puts 'Could not create a new user with given bundle'
```

Comment:
Could you add more error handling here? If you hit this, 1) you don't know why it failed, and 2) it continues on anyway using the first User in the system and asks you which profile you want to use, which is surprising.

Reply:
Alright, so I'll just exit the program then if it fails?

Reply:
That's how it'll be handled now on.